GridTariffAPI.lib/Services/TariffQueryService.cs

```diff
@@ -53,60 +53,109 @@ public class TariffQueryService : ITariffQueryService
+            return retVal;
+        }
 
+        public MeteringPointsAndPriceLevels CheckPriceLevelForMeteringPoints(string fixedPriceId, FixedPriceLevel fixedPriceLevel, List<MeteringPointInformation> meteringPointInformation)
+        {
+            MeteringPointsAndPriceLevels retVal = null;
             var minVal = fixedPriceLevel.ValueMin ?? double.MinValue;
             var maxVal = fixedPriceLevel.ValueMax ?? double.MaxValue;
 
-            var mpInformations = meteringPointInformation.Where(x => minVal <= x.MaxConsumption && x.MaxConsumption < maxVal).ToList();
+            var mpInformations = meteringPointInformation.Where(x => (!fixedPriceLevel.ValueMin.HasValue && !fixedPriceLevel.ValueMax.HasValue)
+                || (x.MaxConsumption.HasValue && minVal <= x.MaxConsumption && x.MaxConsumption < maxVal)).ToList();
             if (mpInformations.Count > 0)
             {
-                retVal = MeteringPointsToPriceLevel(fixedPrice, fixedPriceLevel, mpInformations);
+                retVal = MeteringPointsToPriceLevel(fixedPriceId, fixedPriceLevel.Id, mpInformations);
             }
-            return Task.FromResult(retVal);
+            return retVal;
         }
 
-        public MeteringPointsAndPriceLevels MeteringPointsToPriceLevel(FixedPrices fixedPrice, FixedPriceLevel fixedPriceLevel, List<MeteringPointInformation> mpInformations)
+        public virtual MeteringPointsAndPriceLevels MeteringPointsToPriceLevel(string fixedPriceId, string fixedPriceLevelId, List<MeteringPointInformation> mpInformations)
         {
             var meteringPointAndPriceLevel = new MeteringPointsAndPriceLevels()
             {
                 CurrentFixedPriceLevel = new CurrentFixedPriceLevel(),
-                MeteringPointIds = new MeteringPointIds()
+                MeteringPoints = new MeteringPoints()
             };
 
             foreach (var mpInformation in mpInformations)
             {
-                meteringPointAndPriceLevel.MeteringPointIds.Add(mpInformation.MeteringPointId);
+                meteringPointAndPriceLevel.MeteringPoints.Add(new MeteringPointDetails()
+                {
+                    MeteringPointId = mpInformation.MeteringPointId,
+                    LastUpdated = mpInformation.MaxConsumptionLastUpdated
+                });
             }
-            meteringPointAndPriceLevel.CurrentFixedPriceLevel.Id = fixedPrice.Id;
-            meteringPointAndPriceLevel.CurrentFixedPriceLevel.LevelId = fixedPriceLevel.Id;
-            meteringPointAndPriceLevel.CurrentFixedPriceLevel.LevelValue = null;
-            //TODO LastUpdated should be set per meteringpoint. Change needed in DIGIN.
-            //            meteringPointAndPriceLevel.CurrentFixedPriceLevel.LastUpdated
+            meteringPointAndPriceLevel.CurrentFixedPriceLevel.Id = fixedPriceId;
+            meteringPointAndPriceLevel.CurrentFixedPriceLevel.LevelId = fixedPriceLevelId;
             return meteringPointAndPriceLevel;
         }
```

Comment:
Mora eller dattera igjen... Men kanskje noe sånt?

````
            return new MeteringPointsAndPriceLevels()
            {
                CurrentFixedPriceLevel = new CurrentFixedPriceLevel
                {
                    Id = fixedPriceId,
                    LevelId = fixedPriceLevelId
                },
                MeteringPoints = new MeteringPoints(mpInformations
                    .Select(mpInformation => new MeteringPointDetails()
                    {
                        MeteringPointId = mpInformation.MeteringPointId,
                        LastUpdated = mpInformation.MaxConsumptionLastUpdated
                    })
                    .ToList()))
            };
````
Dette kompilerer ikke. MeteringPoints er et eget objekt, som også er en Collection<MeteringPointDetails>. Kan kanskje løses ved å lage en constructor som tar inn lista (som jeg har basert meg på her)?

Reply:
Ja her er det vel smak og behag. Min subjektive mening er at gjeldende implementasjon er mer lettlest og også enklere å debugge ved evt behov.
Jeg er fristet til å la gjeldende implementasjon stå.
Enig?

Reply:
Det går fint. Jeg synes den foreslåtte varianten er lettere å lese og gjør det tydeligere at vi returnerer en MeteringPointsAndPriceLevels med bl.a. ett objekt per element i lista mpInformation. Men gjeldende variant er oversiktlig nok og løser utfordringen med MeteringPoints vs Collection uten behov for noe ekstra 👍